docs/guides/secrets.md

```diff
@@ -0,0 +1,182 @@
+You could enter those values at block creation, but then they would be saved to the database, and that's what we're trying to avoid.
+
+We need to specify a region in our local AWS config file or in our `AWSCredentials` block.
+The `AwsCredentials` block takes precedence, so let's specify it here for portability.
+
+Under the hood, Prefect is using the AWS `boto3` client to create a session.
+
+Click **Create** to save the blocks.
+
+### Ensure the compute environment has access to AWS credentials
+
+Ensure the compute environment contains AWS credentials with authorization to access AWS Secrets Manager.
+When we connect to Snowflake, Prefect will automatically use these credentials to authenticate and access the secret.
+
+### Create and use `SnowflakeCredentials` and `SnowflakeConnector` blocks in Python code
+
+Let's use Prefect's blocks for convenient access to Snowflake.
+We won't save the blocks, to ensure the credentials are not stored in Prefect Cloud.
+
+We'll create a flow that connects to Snowflake and calls two tasks.
+The first task creates a table and inserts some data.
+The second task reads the data out.
+
+```python
+import json
+from prefect import flow, task
+from prefect_aws import AwsSecret
+from prefect_snowflake import SnowflakeConnector, SnowflakeCredentials
+
+
+@task
+def setup_table(snow_connector: SnowflakeConnector) -> None:
+    with snow_connector as connector:
+        connector.execute(
+            "CREATE TABLE IF NOT EXISTS customers (name varchar, address varchar);"
+        )
+        connector.execute_many(
+            "INSERT INTO customers (name, address) VALUES (%(name)s, %(address)s);",
+            seq_of_parameters=[
+                {"name": "Ford", "address": "Highway 42"},
+                {"name": "Unknown", "address": "Space"},
+                {"name": "Me", "address": "Myway 88"},
+            ],
+        )
+
+
+@task
+def fetch_data(snow_connector: SnowflakeConnector) -> list:
+    all_rows = []
+    with snow_connector as connector:
+        while True:
+            new_rows = connector.fetch_many("SELECT * FROM customers", size=2)
+            if len(new_rows) == 0:
+                break
+            all_rows.append(new_rows)
+    return all_rows
+
+
+@flow(log_prints=True)
+def snowflake_flow():
+    aws_secret_block = AwsSecret.load("my-snowflake-pw")
+    snowflake_pw_json = aws_secret_block.read_secret()
+    snowflake_pw_dict = json.loads(snowflake_pw_json)
+    snowflake_pw_str = snowflake_pw_dict.get("my-snowflake-pw")
+
+    snow_creds = SnowflakeCredentials(
+        role="MYROLE",
+        user="MYUSERNAME",
+        account="ab12345.us-east-2.aws",
+        password=snowflake_pw_str,
+    )
+
+    snow_connector = SnowflakeConnector(
+        schema="MY_SCHEMA",
+        database="MY_DATABASE",
+        warehouse="COMPUTE_WH",
+        fetch_size=1,
+        credentials=snow_creds,
```

Comment:
Putting this nested construction here as an option because it's more my style but totally up to you

```
snow_connector = SnowflakeConnector(
        schema="MY_SCHEMA",
        database="MY_DATABASE",
        warehouse="COMPUTE_WH",
        fetch_size=1,
        credentials=SnowflakeCredentials(
            role="MYROLE",
            user="MYUSERNAME",
            account="ab12345.us-east-2.aws",
            password=snowflake_pw_str,
        ),
        poll_frequency_s=1,
)
```

Reply:
Or even do it all in one shot like
```
aws_secret_block = AwsSecret.load("my-snowflake-pw")

snow_connector = SnowflakeConnector(
        schema="MY_SCHEMA",
        database="MY_DATABASE",
        warehouse="COMPUTE_WH",
        fetch_size=1,
        credentials=SnowflakeCredentials(
            role="MYROLE",
            user="MYUSERNAME",
            account="ab12345.us-east-2.aws",
            password=json.loads(aws_secret_block.read_secret()).get("my-snowflake-pw")
        ),
        poll_frequency_s=1,
)
```
to emphasize the code simplicity of this approach

Reply:
Yeah, I'm a little torn on the step-by-step vs. putting it all together. As this is a more advanced guide, I think one shot with some explanation is probably best. Will update.